### PR TITLE
add feat logsettings system and config

### PR DESCRIPTION
**Pull request by kevinhuy**
## Description

Add new functionality log settings on system and config.
This is done by adding two new class (LogSettingsSystem and LogSettingsConfig) in device.py.

## Motivation and Context

adds a new functionality.

## How Has This Been Tested?

This has been tested on panorama and physical firewall 5220 on version 9.0.10 with python 3.8
below result from panorama.
```
          templates tmpl-common-test;
+          config {
+            shared {
+              log-settings {
+                email {
+                  email_profil {
+                    server {
+                      1.1.1.1 {
+                        display-name myname;
+                        gateway 1.1.1.1;
+                        from myname@name.com;
+                        to tomyname@myname.com;
+                      }
+                    }
+                  }
+                }
+                system {
+                  match-list {
+                    logsystemsystettings_test {
+                      filter "All Logs";
+                      send-email email_profile;
+                      send-snmptrap snmp-profile;
+                      send-syslog srv-profile;
+                      send-http http_profile;
+                    }
+                  }
+                }

```

## Types of changes

- New feature (non-breaking change which adds functionality)

## Checklist

- [ X ] I have updated the documentation accordingly.
- [ X] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes if appropriate.
- [ X ] All new and existing tests passed.

### CODE REVIEW

**Comment by kevinhuy** [comment]
hello :)

@shinmog @btorresgil can you make an review of this MR, feel free to comment .

Thanks

**Comment by btorresgil** [comment]
Thanks for sending this! Will review and merge ASAP.

File: panos/device.py [hunk 1]
```diff
@@ -371,6 +371,114 @@ def _setup(self):
...
+        params.append(
+            VersionedParamPath("send_http", vartype="member", path="send-http")
+        )
+
+        self._params = tuple(params)
+
+
+class LogSettingsConfig(VersionedPanObject):
+    """Firewall or Panorama device log settings configuration
+
```
Comment (shinmog) [inline panos/device.py:430]: Add a note that this is PAN-OS 8.0+

File: panos/device.py [hunk 2]
```diff
@@ -371,6 +371,114 @@ def _setup(self):
...
+    Args:
+        name (string): The name
+        filter (string): Valid values are "All logs" (default) or create your own filter
+        description (string): Description
+        send_to_panorama (bool): Send to panorama
+        send_email (string): Send email profile
+        send_snmp (string): Send snmp profile
+        send_syslog (string): Send syslog profile
+        send_http (string): Send http profile
```
Comment (shinmog) [inline panos/device.py:439]: These are all `vartype="member"`, so the docstring should label them as lists not strings.

File: panos/device.py [hunk 3]
```diff
@@ -371,6 +371,114 @@ def _setup(self):
         self._params = tuple(params)
 
 
+class LogSettingsSystem(VersionedPanObject):
+    """Firewall or Panorama device log settings system
+
```
Comment (shinmog) [inline panos/device.py:376]: Add a note saying this is PAN-OS 8.0+.

File: panos/device.py [hunk 4]
```diff
@@ -371,6 +371,114 @@ def _setup(self):
...
+    Args:
+        name (string): The name
+        filter (string): Valid values are "All logs" (default) or create your own filter
+        description (string): Description
+        send_to_panorama (bool): Send to panorama
+        send_email (string): Send email profile
+        send_snmp (string): Send snmp profile
+        send_syslog (string): Send syslog profile
+        send_http (string): Send http profile
```
Comment (shinmog) [inline panos/device.py:385]: These are `vartype="member"` so should be lists not strings in the docstring.

**Review by shinmog — CHANGES_REQUESTED** [review]
Few actions items

**Comment by kevinhuy** [comment]
@shinmog i have apply the recommandation in your comments , thanks for your help :)

**Review by shinmog — APPROVED** [review]
lgtm

**Pull request merged**